File: 009 Palindrome Number/palindrome_number.py
# LeetCode Submissions by Xinyu Chen
# Palindrome Number 
# https://leetcode.com/problems/palindrome-number/
# Runtime: 480 ms

class Solution:
    # @param {integer} x
    # @return {boolean}
    def isPalindrome(self, x):
        if x == 0:
            return True
        if x < 0:
            return False
        length = math.ceil(math.log10(x + 1))
        base = 10 ** (length - 1)
        for i in range(0, int(length / 2)):
            if x % 10 != x // base:
                return False
            x %= base
            x //= 10
            base //= 100
        return True

File: 009 Palindrome Number/palindrome_number.c
/*	
 *	LeetCode Submissions by Xinyu Chen
 *	Palindrome Number 
 *	https://leetcode.com/problems/palindrome-number/
 *	Runtime: 52 ms
 */
 
bool isPalindrome(long long x) {
	if (x == 0) return true;
	if (x < 0) return false;
	int len = (int)ceil(log10(x + 1));
	long long base = 1;
	for (int i = 0; i < len - 1; i++) base *= 10;
	for (int i = 0; i < len / 2; i++) {
		if (x % 10 != x / base) return false;
		x = x % base;
		x /= 10;
		base /= 100;
	}
	return true;
}